packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -658,6 +661,29 @@ export const RemoteServerFactory = {
                 changeOrigin: true,
                 pathRewrite: {[SLAS_CUSTOM_PROXY_PATH]: ''},
                 onProxyReq: (outGoingReq, incomingReq) => {
+                    // Rewrite key headers.
+                    const newHeaders = rewriteProxyRequestHeaders({
+                        headers: incomingReq.headers,
+                        logging: !isRemote() && verboseProxyLogging,
+                        proxyPath: SLAS_CUSTOM_PROXY_PATH,
+                        targetHost: options.slasHostName,
+                        targetProtocol: 'https'
+                    })
+
+                    Object.entries(newHeaders).forEach(
+                        // setHeader always replaces any current value.
+                        ([key, value]) => outGoingReq.setHeader(key, value)
+                    )
+
+                    Object.keys(incomingReq.headers).forEach((key) => {
+                        // We delete the header on any falsy value, since
+                        // there's no use case where we supply an empty header
+                        // value.
+                        if (!newHeaders[key]) {
+                            outGoingReq.removeHeader(key)
+                        }
+                    })
```

Comment:
This is fine. You might want to consider exporting the original code that you copied so that we don't have this duplication. 

No need for a +1.. since you are merging into your own branch. We'll give it a thorough review once it's ready.

Reply:
Thanks for this input. I did a bit of refactoring to export the original code and remove the code duplication.